lib/libzfs/libzfs_pool.c

```diff
@@ -1756,6 +1756,7 @@ zpool_add(zpool_handle_t *zhp, nvlist_t *nvroot)
 
 	zcmd_write_conf_nvlist(hdl, &zc, nvroot);
 	(void) strlcpy(zc.zc_name, zhp->zpool_name, sizeof (zc.zc_name));
+	zc.zc_cookie = ashift_check;
```

Comment:
I see zc_cookie is already used for many different things in different ioctl's.  Just wonder what will we use for next argument.

Reply:
It probably depends on the argument. We can convert the `ashift_check` to a flag so we can overload the `zc_cookie` or we can add arguments into the nvlist that is part of the `zfs_cmd_t`.

Reply:
> we can add arguments into the nvlist that is part of the zfs_cmd_t.

I haven't looked on that nvlist, but it sounds like less invasive for API/ABI compatibility.

Reply:
I changed this to use the `zc_flags` field in the `zfs_cmd_t` since that seemed more fitting. This can be extended if flags are added at a future date.